flexmeasures/data/models/generic_assets.py

```diff
@@ -262,6 +262,13 @@ def set_attribute(self, attribute: str, value):
 
     def get_consumption_price_sensor(self):
         """Searches for consumption_price_sensor upwards on the asset tree"""
+
+        from flexmeasures.data.models.time_series import Sensor
+
+        if self.consumption_price_sensor_id and not self.consumption_price_sensor:
```

Comment:
So this is something we usually expect SQLAlchemy to do for us, but if the object not loaded through SQLAlchemy we have to establish this property ourselves.

Maybe just add a comment and move on. I can't think of the most elegant solution right now. This seems like a workable approach.